docs/PasswordManagement.md

```diff
@@ -35,32 +38,118 @@ Note: Before the creation of any Splunk deployments in the kubernetes namespace,
+- AWS has created an identity webhook that comes preinstalled in an EKS cluster.
+- This webhook listens to create pod API calls and can inject an additional Token into splunkd pods. This webhook can also be installed into self-managed Kubernetes clusters on AWS using [this guide](https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/SELF_HOSTED_SETUP.md)
+
+Below Example explains the steps required for setting up IAM Service Account
+
+- Follow the steps defined [here to create IAM Role for Service Account](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html)
+
+- Make sure IAM Role only has least amount of privilege necessary for smartstore to work.
+
+- Make sure the service account is used in custom resources where its required
+ Once the Service Account is created, make sure it is annotated with specific IAM Role. Once everything looks good, add service account to splunk custom resource. here is the example for adding it to `Standalone` instance
+
+
+```
+apiVersion: enterprise.splunk.com/v4
+kind: Standalone
+metadata:
+  name: test
+spec:
+  serviceAccount: oidc-service-account
+  smartstore:
+    defaults:
+        volumeName: test-cluster-bucket
+    indexes:
+      - name: main
+        remotePath: $_index_name
+        volumeName: test-cluster-bucket
+      - name: cloudwatch
+        remotePath: $_index_name
+        volumeName: test-cluster-bucket
+    volumes:
+      - name: test-cluster-bucket
+        path: test-cluster-bucket/smartstore
+        endpoint: https://s3-us-west-2.amazonaws.com
+```
+
+- Make sure the IAM service account is used only in required custom resources
+
+- When Splunk pod is running AWS webhook service injects 2 new environment variables `AWS_WEB_IDENTITY_TOKEN_FILE` and `AWS_ROLE_ARN` along with JWS Token file. `splunk` pod reads these environment variables to get temporary AWS credentials from AWS IAM service to access smartstore buckets
+
+***OIDC key management***
+The proper Key management of OIDC is outside of Splunk installation. The customer is responsible to use a properly configured OIDC using certificates from a trusted CA.
+
+***self signed certificate***
+The OIDC should not use self-signed certificates but rather utilize an existing PKI infrastructure, e.g. have the OIDC certificate issued and signed by your organization's CA with proper certificate signature chains and key expiation policies.
+
+***sharing OIDC token file***
+Make sure the token file mentioned in AWS_WEB_IDENTITY_TOKEN_FILE location is only accessible inside of the pod and is not mapped or shared outside of the pod
+
+## Support for AWS IAM Role for Service Account in Splunk Operator Deployment
```

Comment:
Just curious here - This section is not specific to smartstore right? Just for appFramework?

Reply:
OIDC support in splunkd is not specific to smartstore. it can be used for accessing any resource from AWS if the role has permissons to do so and if splunkd required

Reply:
Yes its only for app framework